vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceLinkManagerTest.java

```diff
@@ -226,4 +227,34 @@ public void test_updateLink_success() throws SpaceDoesNotExistException, LinkDoe
         Assert.assertEquals(spaceLinkDisplayUpdated.getLink().getTargetSpace(), actualUpdatedLink.getLink().getTargetSpace());
         Assert.assertEquals(spaceLinkDisplayUpdated.getType(), actualUpdatedLink.getType());
     }
+    
+    @Test
+    public void test_deleteSpaceLinkWithSourceAsNull_present() {
+        SpaceLink spaceLink = new SpaceLink();
+        spaceLink.setSourceSpace(null);
+        spaceLink.setId(spaceLinkId1);
+        List<ISpaceLink> spaceLinks =  new ArrayList<ISpaceLink>();
+        spaceLinks.add(spaceLink);
+        Mockito.when(spaceLinkRepo.findBySourceSpaceIsNull()).thenReturn(spaceLinks);
+        Mockito.doNothing().when(spaceLinkDisplayRepo).deleteByLink(spaceLink); 
+        Mockito.doNothing().when(spaceLinkRepo).deleteBySourceSpaceIdIsNull(); 
+        managerToTest.deleteSpaceLinksWithSourceAsNull();
+    }
+    
+    @Test
+    public void test_deleteSpaceLinkWithSourceAsNull_missing() {
+        ISpace space = new Space();
+        space.setId(spaceId1);
+        SpaceLink spaceLink = new SpaceLink();
+        spaceLink.setSourceSpace(space);
+        spaceLink.setId(spaceLinkId1);
+        List<ISpaceLink> spaceLinks =  new ArrayList<ISpaceLink>();
+        spaceLinks.add(spaceLink);
+        Mockito.when(spaceLinkRepo.findBySourceSpaceIsNull()).thenReturn(spaceLinks);
+        Mockito.doNothing().when(spaceLinkDisplayRepo).deleteByLink(spaceLink); 
+        Mockito.doNothing().when(spaceLinkRepo).deleteBySourceSpaceIdIsNull(); 
+        managerToTest.deleteSpaceLinksWithSourceAsNull();
```

Comment:
I'm confused, what is this testing? The space link has a source space but it is still returned by `Do you want to delete them?`

Reply:
Removed this unit test as it is covered by the previous test

Reply:
Changed this now, as two unit tests covered the same scenario

Reply:
still confused. the method deletes space links with no source space ,right? but the spacelink returned from the mockito call has a source space.